src/msix/PAL/XML/xerces-c/XmlObject.cpp

```diff
@@ -53,20 +128,52 @@ class ParsingException final : public XERCES_CPP_NAMESPACE::ErrorHandler
 
     void warning(const XERCES_CPP_NAMESPACE::SAXParseException& exp) override
     {        
-        ThrowError(MSIX::Error::XmlWarning);
+        ThrowErrorAndLog(MSIX::Error::XmlWarning, GetMessage(exp).c_str());
     }
 
     void error(const XERCES_CPP_NAMESPACE::SAXParseException& exp) override
     {
-        ThrowError(MSIX::Error::XmlError);
+        ThrowErrorAndLog(MSIX::Error::XmlError, GetMessage(exp).c_str());
     }
 
     void fatalError(const XERCES_CPP_NAMESPACE::SAXParseException& exp) override
     {
-        ThrowError(MSIX::Error::XmlFatal);
+        ThrowErrorAndLog(MSIX::Error::XmlFatal, GetMessage(exp).c_str());
     }
 
     void resetErrors() override {}
+private:
+    std::string GetMessage(const XERCES_CPP_NAMESPACE::SAXParseException& exp)
+    {
+        std::u16string utf16FileName = std::u16string(exp.getSystemId());
+        std::u16string utf16Message = std::u16string(exp.getMessage());
+        return "Error " + u16string_to_utf8(utf16Message) + " file: " + u16string_to_utf8(utf16FileName) +
+            ":" + std::to_string(static_cast<std::uint64_t>(exp.getLineNumber())) + "," +
+            std::to_string(static_cast<std::uint64_t>(exp.getColumnNumber()));

```

Comment:
Can we do something more like:

Error in FILENAME.XML [Line 1, Col 1] :: THIS IS AN ERROR MESSAGE